### PR TITLE
[Fix] alter imagepulljob inoperative

### PR DESCRIPTION
**Pull request by ls-2018**
- https://github.com/openkruise/kruise/issues/1386

### CODE REVIEW

**Comment by zmberg** [comment]
Would setting the completionPolicy to never be sufficient for your needs? The ability to do this needs to be discussed, and it shouldn't be possible to implement it this way.

**Comment by ls-2018** [comment]
👌🏻, look forward to the discussion

**Comment by zmberg** [comment]
@ls-2018 Can I talk to you in the dingding group?

**Comment by ls-2018** [comment]
Of course, my name is Liu Shuo

Like k8s native job, it is not allowed to change after completion.
And our crd can be changed, which gives me the feeling that it can be changed and rerun.
Should we do the same or improve on it？

File: pkg/controller/imagepulljob/imagepulljob_controller.go [hunk 1]
```diff
@@ -92,7 +97,19 @@ func add(mgr manager.Manager, r *ReconcileImagePullJob) error {
...
 	// Watch for changes to ImagePullJob
-	err = c.Watch(&source.Kind{Type: &appsv1alpha1.ImagePullJob{}}, &handler.EnqueueRequestForObject{})
+	err = c.Watch(&source.Kind{Type: &appsv1alpha1.ImagePullJob{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldIPJ := e.ObjectOld.(*appsv1alpha1.ImagePullJob)
+			newIPJ := e.ObjectNew.(*appsv1alpha1.ImagePullJob)
+
+			if !apiequality.Semantic.DeepEqual(oldIPJ.Spec, newIPJ.Spec) && oldIPJ.Status.CompletionTime != nil {
+				newIPJ.Status = appsv1alpha1.ImagePullJobStatus{}
+				r.Status().Update(context.TODO(), newIPJ)
```
Comment (veophi) [inline pkg/controller/imagepulljob/imagepulljob_controller.go:107]: Do not write object in event handler...

**Pull request closed without merging**